diff --git a/drivers/of/of_mdio.c b/drivers/of/of_mdio.c
index 5ad1342..a40d570 100644
--- a/drivers/of/of_mdio.c
+++ b/drivers/of/of_mdio.c
@@ -97,8 +97,8 @@ static int of_mdiobus_register_phy(struct mii_bus *mdio,
 		return rc;
 	}
 
-	dev_dbg(&mdio->dev, "registered phy %pOFn at address %i\n",
-		child, addr);
+	dev_info(&mdio->dev, "registered phy %pOFn at address %i with phy_id %x\n",
+		child, addr, phy->phy_id);
 	return 0;
 }
 
